presentation/src/main/java/com/safetyheads/akademiaandroida/presentation/ui/viewmodels/MediaViewModel.kt

```diff
@@ -0,0 +1,74 @@
+package com.safetyheads.akademiaandroida.presentation.ui.viewmodels
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.safetyheads.akademiaandroida.domain.entities.Media
+import com.safetyheads.akademiaandroida.domain.entities.Settings
+import com.safetyheads.akademiaandroida.domain.repositories.SettingsRepository
+import com.safetyheads.akademiaandroida.domain.usecases.GetAllVideoUseCase
+import com.safetyheads.akademiaandroida.domain.usecases.GetInstagramImageUseCase
+import kotlinx.coroutines.launch
+
+class MediaViewModel(
+    private val getInstagramImageUseCase: GetInstagramImageUseCase,
+    private val getAllVideoUseCase: GetAllVideoUseCase,
+    private val settingsRepository: SettingsRepository
+) : ViewModel() {
+
+    init {
+        getAllVideosFromYouTube()
+        getInstagramImage()
+    }
+
+    val youtubeFilmList: MutableLiveData<List<Media>> = MutableLiveData()
+    val instagramImageList: MutableLiveData<List<Media>> = MutableLiveData()
+    val connectedList: MutableLiveData<List<Media>> = MutableLiveData()
```

Comment:
I think https://developer.android.com/reference/android/arch/lifecycle/MediatorLiveData will be exactly what you need here.